src/pages/ApplyMain/index.js

```diff
@@ -1,32 +1,94 @@
-import React, { useEffect, useState } from 'react';
+import React, { useState, useEffect } from 'react';
+
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
 
 import Button from '../../components/Button';
 
-import PartInformation from './PartInformation';
+import ApplyMainTop from './ApplyMainTop';
+import { PART_DATA } from './PartData.js';
+import PartInfo from './PartInfo';
+import PartInfoMobile from './PartInfoMobile';
+import PartInfoTablet from './PartInfoTablet';
 
 const ApplyMainPage = () => {
-  const [testState, setTestState] = useState('');
-  const [error, setError] = useState('');
-  const test = e => {
-    setTestState(e.target.value);
-  };
+  const [innerWidth, setInnerWidth] = useState(window.innerWidth);
+  const DESKTOP_WIDTH = 1199;
+  const TABLET_WIDTH = 599;
 
-  useEffect(() => {
-    if (testState.length >= 5 || testState.length === 0) {
-      setError('에러입니당');
+  const changeWidth = (apply, index) => {
+    if (innerWidth > DESKTOP_WIDTH) {
+      return <PartInfo partInfo={apply} key={index} />;
+    } else if (innerWidth > TABLET_WIDTH) {
+      return <PartInfoTablet partInfo={apply} key={index} />;
     } else {
-      setError('');
+      return <PartInfoMobile partInfo={apply} key={index} />;
     }
-  }, [testState]);
+  };
+
+  useEffect(() => {
+    const resizeListener = () => {
+      setInnerWidth(window.innerWidth);
+    };
+    window.addEventListener('resize', resizeListener);
+  });
```

Comment:
resize에 대해 eventListener 를 달아준건 좋은데, 리액트 useEffect 에서 달아줄 경우에는 이 컴퍼넌트가 언마운트 될 때 저 리스너도 떼어주셔야 합니다. 아래처럼 작성하면 됩니다. 나중에 useEffect cleanup function 에 대해 더 공부해보시면 좋을거에요!

```suggestion
  useEffect(() => {
    const resizeListener = () => {
      setInnerWidth(window.innerWidth);
    };
    window.addEventListener('resize', resizeListener);
    return () => {
      window. removeEventListener('resize', resizeListener);
    };
  });
```

Reply:
미친 코드리뷰 클라스 ..